### DISCUSSION TITLE
Why multiple queries for returning data + count?

### DISCUSSION THREAD

**Discussion by atawfique21** [prisma/prisma#8830] [Q&A] [answered]
I see answers such as https://github.com/prisma/prisma/discussions/8103 and https://github.com/prisma/prisma/discussions/3087 which both essentially recommend executing the two calls separately, in a transaction to return both the data + dataCount. This does work and thank you for shedding light on this @ryands17.

However, my question is: Doesn't it make sense for `count: true` to exist in the context of `___.findMany()` for performance reasons, as we can cut down a query? This seems like an obvious choice. Or is it true that a transaction counts as one total query so there is no inefficiency of using transaction to call two queries? Making the `count: true` call in `findMany()` a bit redundant and only for more readable code? 

If indeed, there are performance benefits to including a `count:` call inside of `findMany()`, this is definitely something a lot of endusers will find useful for pagination, rendering total count of results, and furthermore and would warrant placing it on the roadmap.

Let me know what you think, and please excuse my ignorance on this topic, just trying to learn.

**Comment by ryands17** [comment, marked as answer]
@atawfique21 👋 
Yes currently two queries is the only way. We have a request for the same here that you can add a 👍 to and follow for more updates:
https://github.com/prisma/prisma/issues/7550

If it exists in `findMany`, it would still be 2 queries but yes the roundtrip to the database would just be 1.

**Reply by felinto-dev** [reply]
If I use .$transaction the roundtrip to the database still will be 2?

```ts
    const [items, count] = await this.prisma.$transaction([
      this.prisma.platformAccount.findMany(query),
      this.prisma.platformAccount.count({ where: query.where }),
    ]);

    return { items, count };
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
